skillmap/src/components/AppModal.tsx

```diff
@@ -419,77 +421,126 @@ export class AppModalImpl extends React.Component<AppModalProps, AppModalState>
     renderRewardModal() {
         const { reward } = this.props;
 
-        if (reward?.type === "certificate") return this.renderCertificateModal(reward);
-        else if (reward?.type === "completion-badge") return this.renderBadgeModal(reward);
+        let modal: JSX.Element | undefined;
+
+        if (reward?.type === "certificate") modal = this.renderCertificateModal(reward);
+        else if (reward?.type === "completion-badge") modal = this.renderBadgeModal(reward);
+
+        return <div className="confetti-container">
+            {modal}
+            {this.renderConfetti()}
+        </div>
     }
 
     renderCertificateModal(reward: MapRewardCertificate) {
         const title = lf("Rewards");
         const  { mapId, skillMap, activity, hasPendingModals, dispatchNextModal } = this.props;
 
-        const buttons: ModalAction[] = [];
-        buttons.push({
-            label: lf("Open Certificate"),
+        const buttons: JSX.Element[] = [];
 
-            onClick: () => {
-                tickEvent("skillmap.openCertificate", { path: mapId, activity: activity!.activityId });
-                window.open((reward as MapRewardCertificate).url || skillMap!.completionUrl);
-            }
-        })
+        const onCertificateClick = () => {
+            tickEvent("skillmap.openCertificate", { path: mapId, activity: activity!.activityId });
+            window.open((reward as MapRewardCertificate).url || skillMap!.completionUrl);
+        };
+
+        buttons.push(
+            <button key="cert" className="completion-reward inverted right-icon" onClick={onCertificateClick}>
+                <span>{lf("Certificate")}</span>
+                <i className="icon file outline" />
+            </button>
+        )
 
         if (hasPendingModals) {
-            buttons.push({
-                label: lf("Next Reward"),
-                onClick: () => {
-                    tickEvent("skillmap.nextReward", { path: mapId, activity: activity!.activityId, currentReward: reward!.type });
-                    dispatchNextModal();
-                }
-            })
+            const onNextRewardClick = () => {
+                tickEvent("skillmap.nextReward", { path: mapId, activity: activity!.activityId, currentReward: reward!.type });
+                dispatchNextModal();
+            };
+
+            buttons.push(
+                <button key="next" className="completion-reward right-icon" onClick={onNextRewardClick}>
+                    <span>{lf("Next Reward")}</span>
+                    <i className="icon right circle arrow" />
+                </button>
+            )
         }
 
-        return <Modal title={title} actions={buttons} onClose={this.handleOnClose}>
+        return <Modal title={title} onClose={this.handleOnClose}>
             {lf("Use the button below to get your completion certificate!")}
+            <div className="reward-buttons">
```

Comment:
are we doing some special/different design for the reward modal buttons? can we make these changes across all the skillmap modal buttons? don't love moving them out like this, "open reward link in new tab" and "move to the next modal" both still feel like modal actions to me

Reply:
We were already doing that for the completion modal, I'm just reusing the css from the button in that modal

Reply:
the completion modal also has separate follow-up actions though, these are the follow-up actions for this modal. this is more a design question than an implementation one, i guess. could we pass in the modal button class as a property to the actions?

Reply:
![2021-11-09 10 11 05](https://user-images.githubusercontent.com/13754588/140980544-ffaa2ac2-8ba6-452a-b268-10de19a6ce8d.gif)